qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDefinition.java

```diff
@@ -270,6 +273,37 @@ private CompletableFuture<List<? extends LocationLink>> findDefinitionFromPart(P
 		}
 	}
 
+	/**
+	 * Find parameter definition for Enum in #when section.
+	 *
+	 * @param parameter
+	 * @param template
+	 * @param cancelChecker
+	 *
+	 * @return parameter definition for Enum in #when section.
+	 */
+	private CompletableFuture<List<? extends LocationLink>> findDefinitionFromParameter(Parameter parameter,
+			Template template, CancelChecker cancelChecker) {
+		String projectUri = template.getProjectUri();
+		Section whenSection = parameter.getParentSection().getParentSection();
+		if (whenSection != null && whenSection.getSectionKind() == SectionKind.WHEN) {
+			// Ex: {#when Machine.status}
+			// {#is ON }
+			// {#is OF|F}
+			// {/when}
+			Parameter value = ((BaseWhenSection) whenSection).getValueParameter();
+			ResolvedJavaTypeInfo whenSwitchJavaTypeInfo = javaCache.resolveJavaType(value, projectUri).getNow(null);

```

Comment:
Rename whenSwitchJavaTypeInfo with whenJavaType

Reply:
You need to check that whenJavaType is an enum. It should now work for instance with String type.
You could write a test with switch which have a type of String and have a case with à string field. When définition will be done it should find the java définition.